src/fn/edit-contacts.js

```diff
@@ -0,0 +1,169 @@
+      info('Processing CSV file:', csv, '…');
+
+
+      const nameParts = fs.path.basename(csv).split('.');
+      const prefix = nameParts[0];
+      info('Processing CSV file:', prefix, '…');
+
+      switch(prefix) {
+        case 'contact':  return downloadDocs(csv, getIDs(csv),db, args);
+        default: throw new Error(`Unrecognised CSV type ${prefix} for file ${csv}`);
+      }
+    })
+    .then(docs => addToModel(csv, docs)),
+    Promise.resolve())
+
+  .then(() => model.exclusions.forEach(toDocs.removeExcludedField))
+  .then(() => Promise.all(Object.values(model.docs).map(saveJsonDoc)));
+};
+
+const model = {
+  csvFiles: {},
+  docs: {},
+  exclusions: [],
+  users: []
+};
+
+const addToModel = (csvFile, docs) => {
+    csvFile = csvFile.match(/^(?:.*[\/\\])?csv[\/\\](.*)\.csv$/)[1]; // eslint-disable-line no-useless-escape
+    model.csvFiles[csvFile] = docs;
+    docs.forEach(doc => {
+      model.docs[doc._id] = doc;
+    });
+  };
+
+function getIDs(csv) {
+    const { rows, cols } = fs.readCsv(csv);
+    var index = cols.indexOf('uuid');
+    if (index === -1){
+     throw Error('missing "uuid" column.');
+   }
+   return rows.map((item) => item[index]);
+}
+
+function processContacts(contactType, csv, ids, contactDocs, args){
+  const { rows, cols } = fs.readCsv(csv);
+  var colNames = args.colNames;
+  var index = cols.indexOf('uuid');
+  var toIncludeIndex = [];
+  if (colNames.length === 0) {
+    warn(' No columns specified, the script will add all the columns in the CSV!');
+    colNames = cols;
+  } else {
+    var splitFileColumns = [];
+    var fileColumn;
+    for(fileColumn of cols) {
+      if (fileColumn !== null) {
+        splitFileColumns.push(fileColumn.split(':')[0]);
+      }      
+    }
+    const columnsValid = colNames.every(function(column) { return splitFileColumns.includes(column);});
+    if (!columnsValid){
+      throw Error('The column name(s) specified do not exist.');
+    }
+
+    colNames = cols.filter(e => colNames.includes(e.split(':')[0]));
+    colNames.forEach(function(e){
+      toIncludeIndex.push(cols.indexOf(e));
+    });
+
+  }
+  if (colNames.includes('uuid')) {
+    colNames.splice(colNames.indexOf('uuid'),1);
+  }
+  return rows
+  .map(r => processCsv(contactType, colNames, r, ids, index, toIncludeIndex, contactDocs));
+
+}
+
+function processCsv(docType, columnNames, rowItem, ids, index, toIncludeIndex, contactDocs) {
+ var row = [];
+ var cols = columnNames;
+ const contactId = rowItem[index];
+ const doc = contactDocs[contactId];
+
+ if(toIncludeIndex.length > 0){
+  toIncludeIndex.forEach(function(index){
+    row.push(rowItem[index]);
+  });
+ } else {
+  rowItem.splice(index,1);
+  row=rowItem;
+ }
+
+ for(let i=0; i<cols.length; ++i) {
+  const { col, val, excluded } = toDocs.parseColumn(cols[i], row[i]);
+  toDocs.setCol(doc, col, val);
+    if(excluded) model.exclusions.push({
+      doc: doc,
+      propertyName: col,
+    });
+  }
+  return doc;
+}
+
+const fetch = {
```

Comment:
Were you planning to add more properties to fetch? otherwise, the `fetchContactList` function can be standalone.

Reply:
Yeah, we were planning to also process user-settings documents, but we've opted to use reuse the code. We've also made the `fetchContactList ` to be standalone since we opted to reuse it.